core/solver/idr.cpp

```diff
@@ -178,6 +179,7 @@ void Idr<ValueType>::iterate(const LinOp *b, LinOp *x) const
         if (stop_criterion->update()
                 .num_iterations(total_iter)
                 .residual(residual.get())
+                .residual_norm(residual_norm.get())
```

Comment:
Why is this better? You had to add an extra `compute_norm2` anyway.

Reply:
No, that norm added is outside the loop. And that was so that a valid value is passed in the first iteration. In the loop, a residual_norm was already being computed in every iteration.